### DISCUSSION TITLE
Are port updates to outdated package versions possible and welcome?

### DISCUSSION THREAD

**Discussion by WKarel** [microsoft/vcpkg#34241] [Q&A] [answered]
I'd like to build Python 3.10 (on Windows x64 using MSVC 2019), while my built-in baseline has its Python version set to [3.11](https://github.com/microsoft/vcpkg/blob/38dd6b15ada368740b5047e4282cbd8395228f03/versions/baseline.json#L6680), already. I expected Python 3.10 to build nevertheless, using the following manifest:
`
{
  "dependencies": [ "python3" ],
  "overrides": [ { "name": "python3", "version": "3.10.7#7" } ],
  "builtin-baseline": "38dd6b15ada368740b5047e4282cbd8395228f03"
}
` 
However, this fails, because libffi's port [has been changed](https://github.com/microsoft/vcpkg/commit/edd3db2e3be3967073756cc83cd500135eaaae69) to produce "ffi.lib" insted of "libffi.lib" on Windows since the most recent port version (`#7`) for Python 3.10 has been released. A simple fix would be to apply the [changes to the port for Python 3.11](https://github.com/microsoft/vcpkg/commit/edd3db2e3be3967073756cc83cd500135eaaae69#diff-f682e6e46ef0d0dae2ceba81d191975da58a84cbdab4811a0e84683442265461) in that same commit also to port `#7` for Python 3.10, and commit that as port `#8` for Python 3.10.

But it seems to me that a minimalistic manifest
`{
  "dependencies": [ "python3" ]
}
`
without explicit baseline and version override simply uses what is there in ports/python3, instead of checking out the according git tree into buildtrees/versioning_/versions/python3/<hash>, as is done when using version overrides. So even for Python 3.11+, the revision of the port for Python 3.10 would be used.

Is it possible at all to update ports of "outdated" package versions? If so, how? Would a pull request be welcome?

Note that I am aware of alternatives to successfully build Python 3.10 (lower baseline and/or version override also for libffi).

**Comment by autoantwort** [comment, marked as answer]
> Is it possible at all to update ports of "outdated" package versions? If so, how? 

No there is currently no good story to do that.

It should be possible by (not tested):
1. checkout old version and create new branch. 
2. Make changes, commit, run `x-add-version` and commit. 
3. Merge with master and discard the new changes in `ports/<port>` and `baseline.json`.
4. Merge new branch into master (should be possible to create a new PR)

One downside is that there is no baseline commit that can be used to select the version, so you have to use a version override. 
You can also leave point 3, but then you have to use a baseline and it can be confusing when the version in the baseline is not the version in the `ports` dir.

**Reply by WKarel** [reply]
Thanks, I've created an according [issue](https://github.com/microsoft/vcpkg/issues/34259).
